opm/simulators/wells/WellGroupHelpers.cpp

```diff
@@ -217,29 +212,48 @@ setCmodeGroup(const Group& group,
                 continue;
 
             const auto& controls = group.injectionControls(phase, summaryState);
-            group_state.injection_control(group.name(), phase, controls.cmode);
+            group_state.update_injection_control(group.name(), phase, controls.cmode);
         }
     }
 
     if (group.isProductionGroup()
         && events.hasEvent(group.name(), ScheduleEvents::GROUP_PRODUCTION_UPDATE)) {
         const auto controls = group.productionControls(summaryState);
-        group_state.production_control(group.name(), controls.cmode);
+        if (controls.cmode == Group::ProductionCMode::FLD && group.name() != "FIELD") {
+            // Set control mode to be the one of the most immediate parent group
+            // with an actual (not NONE or FLD) mode set.
+            auto pname = group.parent();
```

Comment:
It looks nice until we test otherwise. But will we guarantee that we can get something with 
```c++
           auto parent_ctrl = group_state.production_control(pname);
```
Maybe we should have a backup plan to use when the things in the `if` condition does not result in a `parent_ctrl`. 
```c++
group_state.update_production_control(group.name(), controls.cmode);
```